src/Sparrow/LowMemory/LowMemoryNotification.cs

```diff
@@ -313,8 +314,9 @@ private int CheckMemoryStatus()
 
             var memInfo = MemoryInformation.GetMemoryInfo();
 
-            var currentProcessMemoryMappedShared = GetCurrentProcessMemoryMappedShared();
-            var availableMem = (memInfo.AvailableMemory + currentProcessMemoryMappedShared);
+        var result = new SmapsReader().CalculateMemUsageFromSmaps<SmapsReaderNoAllocResults>();
```

Comment:
Can we avoid this allocation here?

Reply:
we can hold static SmapsReader instance in MemoryInformation.

Reply:
Would that be thread safe, though?

Reply:
it is being called from the LowMemoryNotification thread.  We have only one..

Reply:
Then let's allocate that on the method, and pass as argument, rather than a static fifeld.